Sources/App/RouteHandlers/APIRouteHandlers.swift

```diff
@@ -142,7 +142,7 @@ internal class APIHandlers {
 
                     """
                 }
-                return contents
+                return contents.replacingOccurrences(of: "Icon\r\n", with: "Icon\r\r\n", options: .regularExpression)
```

Comment:
This is a general api handler and I think we should not add handler for a specific `macOS` template.

Isn't that because of invalid new line character (`^M`) defined here? https://github.com/toptal/gitignore/blob/master/templates/macOS.gitignore

I have not tried running this branch yet, will check later.

Reply:
The issue is that the controller replaces multiple `\r`s, which is fine but this is an edge-case where we don't want it.